### PR TITLE
feat(web): Add action button to search result page

### LINKED ISSUE #2307: [Feature] Add actions to app bar for search result
### Discussed in https://github.com/immich-app/immich/discussions/2114

<div type='discussions-op-text'>

<sup>Originally posted by **JVT038** March 29, 2023</sup>
### The feature

When I select results from a search query, I can't do anything with them. 
I can't add them to an album, download them, share them, or delete them.
See web interface example for what I'd like in the search bar on web:
image

On mobile, I can't select any items at all.
So I'd like to be able to select items on the search page and do something with them, just like I can select items on the main timeline and do something.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile</div>

### PR DESCRIPTION
**Pull request by alextran1502**
Resolved #2307 #2308

* Add archive icon to thumbnail - only displayed on the search page
* Favorite and archive icon shows on the thumbnail on the search page
* Bulk toggle archive/favorite - if one of the selected assets is not in the archive/favorite, the action would be to archive/favorite all - if all the selected assets are archived/favorite or not-archived/not-favorite, the option would be unarchived/unfavorite and archived/favorite, respectively

### CODE REVIEW

File: web/src/lib/components/assets/thumbnail/thumbnail.svelte [hunk 1]
```diff
@@ -114,6 +116,11 @@
 					</div>
 				{/if}
 
+				{#if showArchiveIcon && asset.isArchived}
+					<div class="absolute {asset.isFavorite ? 'bottom-10' : 'bottom-2'} left-2 z-10">
+						<ArchiveArrowDownOutline size="24" class="text-white" />
+					</div>
+				{/if}
```
Comment (michelheusschen) [inline web/src/lib/components/assets/thumbnail/thumbnail.svelte:123]: Haven't looked into what's causing this yet, but the icons can sometimes show incorrectly:
1. Favorite and archive an asset and the icons show on top of each other
2. Have an asset that is favorited + archived (reload page), then remove from favorites and the archived icon doesn't properly align

I think the icons look better stacked horizontally, but that's a preference thing

File: web/src/routes/(user)/search/+page.svelte [hunk 2]
```diff
@@ -23,14 +41,193 @@
...
+					title="Share"
+					logo={ShareVariantOutline}
+					on:click={handleCreateSharedLink}
+				/>
+
+				<CircleIconButton
+					title="Unarchive"
+					logo={isAllArchived ? ArchiveArrowUpOutline : ArchiveArrowDownOutline}
+					on:click={toggleArchive}
+				/>
```
Comment (michelheusschen) [inline web/src/routes/(user)/search/+page.svelte:209]: Title should be `Archive` or `Unarchive`

**Review by michelheusschen — COMMENTED** [review]
Looks fine in general, but I've added some comments. We should also try to avoid the code duplication for the action buttons, but that's probably outside the scope of this PR.

File: web/src/lib/components/assets/thumbnail/thumbnail.svelte (hunk 1 above)
Comment (alextran1502) [reply web/src/lib/components/assets/thumbnail/thumbnail.svelte:123]: Yeah, I can reproduce, look like the class isn't updated correctly

Comment (alextran1502) [reply web/src/lib/components/assets/thumbnail/thumbnail.svelte:123]: resolved by updating the state of the assets that get passed in GalleryViewer

**Pull request merged**